stl/inc/iterator

```diff
@@ -1229,18 +1231,30 @@ public:
     }
 
     constexpr decltype(auto) operator++(int) {
+        class _Length_guard {
+        public:
+            explicit _Length_guard(counted_iterator* _Target_) noexcept : _Target(_Target_) {}
+
+            _Length_guard(const _Length_guard&)            = delete;
+            _Length_guard& operator=(const _Length_guard&) = delete;
+
+            ~_Length_guard() {
+                if (_STD uncaught_exceptions() > _Exceptions) {

```

Comment:
I'm afraid that this will cause worse codegen, because compilers don't seems able to fold `_STD uncaught_exceptions() > _Exceptions` to `false` when no exception is thrown. Perhaps this is also a reason for not standardizing `scope_fail`.

Reply:
I agree. Perhaps #2307 is complete then. if we cannot do anything about the remaining `<vector>` occurrences.